version.go

```diff
@@ -0,0 +1,41 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	// BuildDate is date when binary was built
+	BuildDate string
+	// BuildVersion is the version of binary
+	BuildVersion string
+)
+
+// providerVersion holds current provider version
+type providerVersion struct {
+	Version string `json:"version"`
+	Date    string `json:"date"`
+	// MinDriverVersion is minimum driver version the provider works with
+	MinDriverVersion string `json:"minDriverVersion"`
+}
+
+func printVersionAndExit() {
+	pv := providerVersion{
+		Version:          BuildVersion,
+		Date:             BuildDate,
+		MinDriverVersion: minDriverVersion,
+	}
+
+	var res []byte
+	var err error
+	if res, err = json.Marshal(pv); err != nil {
+		log.Fatalf("[error]: %v", err)
+	}
+
+	fmt.Printf(string(res) + "\n")
+	os.Exit(0)
```

Comment:
nit: in general we should return err and check for error or success in main and have main do the exit

Reply:
done!